conbench/entities/benchmark_result.py

```diff
@@ -138,168 +142,116 @@ def create(userres):
         validate_and_augment_result_tags(userres)
         validate_run_result_consistency(userres)
 
-        result_error_for_db: Optional[Dict] = None
-
-        # Handle 'error' scenarios. Obvious: user-given error object set.
+        # The dict that is used for DB insertion later, populated below.
+        result_data_for_db: Dict = {}
+
+        if "stats" in userres:
+            # First things first: use the complete user-given `stats` object
+            # for potential DB insertion down below. In `error` state, do not
+            # perform deeper validation of the user-given stats object (the
+            # benchmark result is not used for any kind of analysis, which is
+            # why it's probably ok to store the user-given 'stats' object w/o
+            # deeper validation, maybe the data is helpful for debugging). Note
```

Comment:
This is essentially a philosophical decision, and a related ticket is
https://github.com/conbench/conbench/issues/395